File: app/src/main/java/com/neome/java/api/home/main/RpcMain.java
// neome.ai API - do not change
//
/* eslint-disable @typescript-eslint/no-unused-vars */
// noinspection UnusedGlobalSymbols,ES6UnusedImports,JSUnusedGlobalSymbols,JSUnusedLocalSymbols

package com.neome.java.api.home.main;

import com.neome.java.api.core.base.msg.MsgEntUserIdNoVersion;
import com.neome.java.api.home.base.msg.MsgChatId;
import com.neome.java.api.home.base.msg.MsgGroupId;
import com.neome.java.api.home.main.msg.MsgChatBlock;
import com.neome.java.api.home.main.msg.MsgMessageBulkGet;
import com.neome.java.api.home.main.msg.MsgMessageEdit;
import com.neome.java.api.home.main.msg.MsgMessageForwardCandidateList;
import com.neome.java.api.home.main.msg.MsgMessageList;
import com.neome.java.api.home.main.msg.MsgMessageListJump;
import com.neome.java.api.home.main.msg.MsgMessageListOffset;
import com.neome.java.api.home.main.msg.MsgMessageReport;
import com.neome.java.api.home.main.msg.MsgMessageSend;
import com.neome.java.api.home.main.msg.MsgOffset;
import com.neome.java.api.home.main.msg.MsgOffsetWithVersion;
import com.neome.java.api.home.main.msg.MsgReaction;
import com.neome.java.api.home.main.msg.MsgReverseGeocode;
import com.neome.java.api.home.main.msg.MsgUrl;
import com.neome.java.api.home.main.sig.SigChatCandidateMap;
import com.neome.java.api.home.main.sig.SigGroupMessageCandidateList;
import com.neome.java.api.home.main.sig.SigLinkPreview;
import com.neome.java.api.home.main.sig.SigMessage;
import com.neome.java.api.home.main.sig.SigMessageBulk;
import com.neome.java.api.home.main.sig.SigMessageList;
import com.neome.java.api.home.main.sig.SigReverseGeocode;
import com.neome.java.api.home.main.sig.SigUserLastOnline;
import com.neome.java.api.home.main.sig.SigUserMessageCandidateList;
import com.neome.java.api.meta.base.Types.EntId;
import com.neome.java.api.meta.base.Types.ServiceName;
import com.neome.java.api.nucleus.base.ApiPlus;
import com.neome.java.api.nucleus.base.CallFactory;
import com.neome.java.api.nucleus.base.ISigAcceptor;
import com.neome.java.api.nucleus.base.sig.SigDone;

@SuppressWarnings("unused")
public class RpcMain
{
  public static final ServiceName SN = ServiceName.main;

  public static void chatBlock(MsgChatBlock msg, ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, ApiPlus.ENT_ID_GLOBAL, SN, "chatBlock")
      .sendBearerToken()
      .put(msg, sigAcceptor);
  }

  public static void chatClear(EntId entId, MsgChatId msg, ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "chatClear")
      .sendBearerToken()
      .delete(msg, sigAcceptor);
  }

  public static void chatRemove(EntId entId, MsgChatId msg, ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "chatRemove")
      .sendBearerToken()
      .delete(msg, sigAcceptor);
  }

  public static void groupJoin(MsgGroupId msg, ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, ApiPlus.ENT_ID_GLOBAL, SN, "groupJoin")
      .sendBearerToken()
      .put(msg, sigAcceptor);
  }

  public static void groupMessageCandidateListGet(EntId entId,
    ISigAcceptor<SigGroupMessageCandidateList> sigAcceptor)
  {
    CallFactory.rpc.create(SigGroupMessageCandidateList.class, entId, SN,
        "groupMessageCandidateListGet")
      .sendBearerToken()
      .get(null, sigAcceptor);
  }

  public static void linkPreviewGet(MsgUrl msg, ISigAcceptor<SigLinkPreview> sigAcceptor)
  {
    CallFactory.rpc.create(SigLinkPreview.class, ApiPlus.ENT_ID_GLOBAL, SN, "linkPreviewGet")
      .sendBearerToken()
      .get(msg, sigAcceptor);
  }

  public static void messageBulkGet(EntId entId, MsgMessageBulkGet msg,
    ISigAcceptor<SigMessageBulk> sigAcceptor)
  {
    CallFactory.rpc.create(SigMessageBulk.class, entId, SN, "messageBulkGet")
      .sendBearerToken()
      .post(msg, sigAcceptor);
  }

  public static void messageEdit(EntId entId, MsgMessageEdit msg, ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "messageEdit")
      .sendBearerToken()
      .patch(msg, sigAcceptor);
  }

  public static void messageForwardCandidateListGet(EntId entId, MsgMessageForwardCandidateList msg,
    ISigAcceptor<SigChatCandidateMap> sigAcceptor)
  {
    CallFactory.rpc.create(SigChatCandidateMap.class, entId, SN, "messageForwardCandidateListGet")
      .sendBearerToken()
      .get(msg, sigAcceptor);
  }

  public static void messageGet(EntId entId, MsgOffsetWithVersion msg,
    ISigAcceptor<SigMessage> sigAcceptor)
  {
    CallFactory.rpc.create(SigMessage.class, entId, SN, "messageGet")
      .sendBearerToken()
      .get(msg, sigAcceptor);
  }

  public static void messageListGet(EntId entId, MsgMessageList msg,
    ISigAcceptor<SigMessageList> sigAcceptor)
  {
    CallFactory.rpc.create(SigMessageList.class, entId, SN, "messageListGet")
      .sendBearerToken()
      .get(msg, sigAcceptor);
  }

  public static void messageListJump(EntId entId, MsgMessageListJump msg,
    ISigAcceptor<SigMessageList> sigAcceptor)
  {
    CallFactory.rpc.create(SigMessageList.class, entId, SN, "messageListJump")
      .sendBearerToken()
      .get(msg, sigAcceptor);
  }

  public static void messageListNext(EntId entId, MsgMessageListOffset msg,
    ISigAcceptor<SigMessageList> sigAcceptor)
  {
    CallFactory.rpc.create(SigMessageList.class, entId, SN, "messageListNext")
      .sendBearerToken()
      .get(msg, sigAcceptor);
  }

  public static void messageListPrev(EntId entId, MsgMessageListOffset msg,
    ISigAcceptor<SigMessageList> sigAcceptor)
  {
    CallFactory.rpc.create(SigMessageList.class, entId, SN, "messageListPrev")
      .sendBearerToken()
      .get(msg, sigAcceptor);
  }

  public static void messageMarkRead(EntId entId, MsgOffset msg, ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "messageMarkRead")
      .sendBearerToken()
      .put(msg, sigAcceptor);
  }

  public static void messageMarkReceived(EntId entId, MsgOffset msg,
    ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "messageMarkReceived")
      .sendBearerToken()
      .put(msg, sigAcceptor);
  }

  public static void messageReactionPut(EntId entId, MsgReaction msg,
    ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "messageReactionPut")
      .sendBearerToken()
      .put(msg, sigAcceptor);
  }

  public static void messageRemoveForEveryone(EntId entId, MsgOffset msg,
    ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "messageRemoveForEveryone")
      .sendBearerToken()
      .delete(msg, sigAcceptor);
  }

  public static void messageRemoveForMe(EntId entId, MsgOffset msg,
    ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "messageRemoveForMe")
      .sendBearerToken()
      .delete(msg, sigAcceptor);
  }

  public static void messageReport(EntId entId, MsgMessageReport msg,
    ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "messageReport")
      .sendBearerToken()
      .post(msg, sigAcceptor);
  }

  public static void messageSend(EntId entId, MsgMessageSend msg, ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "messageSend")
      .sendBearerToken()
      .post(msg, sigAcceptor);
  }

  public static void messageTyping(EntId entId, MsgChatId msg, ISigAcceptor<SigDone> sigAcceptor)
  {
    CallFactory.rpc.create(SigDone.class, entId, SN, "messageTyping")
      .sendBearerToken()
      .post(msg, sigAcceptor);
  }

  public static void reverseGeocode(EntId entId, MsgReverseGeocode msg,
    ISigAcceptor<SigReverseGeocode> sigAcceptor)
  {
    CallFactory.rpc.create(SigReverseGeocode.class, entId, SN, "reverseGeocode")
      .sendBearerToken()
      .get(msg, sigAcceptor);
  }

  public static void userLastOnlineGet(EntId entId, MsgEntUserIdNoVersion msg,
    ISigAcceptor<SigUserLastOnline> sigAcceptor)
  {
    CallFactory.rpc.create(SigUserLastOnline.class, entId, SN, "userLastOnlineGet")
      .sendBearerToken()
      .get(msg, sigAcceptor);
  }

  public static void userMessageCandidateListGet(
    ISigAcceptor<SigUserMessageCandidateList> sigAcceptor)
  {
    CallFactory.rpc.create(SigUserMessageCandidateList.class, ApiPlus.ENT_ID_GLOBAL, SN,
        "userMessageCandidateListGet")
      .sendBearerToken()
      .get(null, sigAcceptor);
  }
}

File: app/src/main/java/com/neome/api/home/main/RpcMain.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.home.main

import com.neome.api.nucleus.base.ApiPlus
import com.neome.api.nucleus.base.CallFactory
import com.neome.api.nucleus.base.ApiPlus
import com.neome.api.meta.base.Types.EntId
import com.neome.api.nucleus.base.ISigAcceptor
import com.neome.api.home.main.msg.MsgChatBlock
import com.neome.api.home.base.msg.MsgChatId
import com.neome.api.core.base.msg.MsgEntUserIdNoVersion
import com.neome.api.home.base.msg.MsgGroupId
import com.neome.api.home.main.msg.MsgMessageBulkGet
import com.neome.api.home.main.msg.MsgMessageEdit
import com.neome.api.home.main.msg.MsgMessageForwardCandidateList
import com.neome.api.home.main.msg.MsgMessageList
import com.neome.api.home.main.msg.MsgMessageListJump
import com.neome.api.home.main.msg.MsgMessageListOffset
import com.neome.api.home.main.msg.MsgMessageReport
import com.neome.api.home.main.msg.MsgMessageSend
import com.neome.api.home.main.msg.MsgOffset
import com.neome.api.home.main.msg.MsgOffsetWithVersion
import com.neome.api.home.main.msg.MsgReaction
import com.neome.api.home.main.msg.MsgReverseGeocode
import com.neome.api.home.main.msg.MsgUrl
import com.neome.api.meta.base.Types.ServiceName
import com.neome.api.home.main.sig.SigChatCandidateMap
import com.neome.api.nucleus.base.sig.SigDone
import com.neome.api.home.main.sig.SigGroupMessageCandidateList
import com.neome.api.home.main.sig.SigLinkPreview
import com.neome.api.home.main.sig.SigMessage
import com.neome.api.home.main.sig.SigMessageBulk
import com.neome.api.home.main.sig.SigMessageList
import com.neome.api.home.main.sig.SigReverseGeocode
import com.neome.api.home.main.sig.SigUserLastOnline
import com.neome.api.home.main.sig.SigUserMessageCandidateList

class RpcMain {
    companion object {
        val SN: ServiceName = ServiceName.main

        fun chatBlock(msg: MsgChatBlock, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, ApiPlus.ENT_ID_GLOBAL, SN, "chatBlock")
                .sendBearerToken()
                .put(msg, sigAcceptor)
        }

        fun chatClear(entId: EntId, msg: MsgChatId, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "chatClear")
                .sendBearerToken()
                .delete(msg, sigAcceptor)
        }

        fun chatRemove(entId: EntId, msg: MsgChatId, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "chatRemove")
                .sendBearerToken()
                .delete(msg, sigAcceptor)
        }

        fun groupJoin(msg: MsgGroupId, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, ApiPlus.ENT_ID_GLOBAL, SN, "groupJoin")
                .sendBearerToken()
                .put(msg, sigAcceptor)
        }

        fun groupMessageCandidateListGet(
            entId: EntId,
            sigAcceptor: ISigAcceptor<SigGroupMessageCandidateList>
        ) {
            CallFactory.rpc.create(
                SigGroupMessageCandidateList::class.java,
                entId,
                SN,
                "groupMessageCandidateListGet"
            )
                .sendBearerToken()
                .get(null, sigAcceptor)
        }

        fun linkPreviewGet(msg: MsgUrl, sigAcceptor: ISigAcceptor<SigLinkPreview>) {
            CallFactory.rpc.create(
                SigLinkPreview::class.java,
                ApiPlus.ENT_ID_GLOBAL,
                SN,
                "linkPreviewGet"
            )
                .sendBearerToken()
                .get(msg, sigAcceptor)
        }

        fun messageBulkGet(
            entId: EntId,
            msg: MsgMessageBulkGet,
            sigAcceptor: ISigAcceptor<SigMessageBulk>
        ) {
            CallFactory.rpc.create(SigMessageBulk::class.java, entId, SN, "messageBulkGet")
                .sendBearerToken()
                .post(msg, sigAcceptor)
        }

        fun messageEdit(entId: EntId, msg: MsgMessageEdit, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "messageEdit")
                .sendBearerToken()
                .patch(msg, sigAcceptor)
        }

        fun messageForwardCandidateListGet(
            entId: EntId,
            msg: MsgMessageForwardCandidateList,
            sigAcceptor: ISigAcceptor<SigChatCandidateMap>
        ) {
            CallFactory.rpc.create(
                SigChatCandidateMap::class.java,
                entId,
                SN,
                "messageForwardCandidateListGet"
            )
                .sendBearerToken()
                .get(msg, sigAcceptor)
        }

        fun messageGet(
            entId: EntId,
            msg: MsgOffsetWithVersion,
            sigAcceptor: ISigAcceptor<SigMessage>
        ) {
            CallFactory.rpc.create(SigMessage::class.java, entId, SN, "messageGet")
                .sendBearerToken()
                .get(msg, sigAcceptor)
        }

        fun messageListGet(
            entId: EntId,
            msg: MsgMessageList,
            sigAcceptor: ISigAcceptor<SigMessageList>
        ) {
            CallFactory.rpc.create(SigMessageList::class.java, entId, SN, "messageListGet")
                .sendBearerToken()
                .get(msg, sigAcceptor)
        }

        fun messageListJump(
            entId: EntId,
            msg: MsgMessageListJump,
            sigAcceptor: ISigAcceptor<SigMessageList>
        ) {
            CallFactory.rpc.create(SigMessageList::class.java, entId, SN, "messageListJump")
                .sendBearerToken()
                .get(msg, sigAcceptor)
        }

        fun messageListNext(
            entId: EntId,
            msg: MsgMessageListOffset,
            sigAcceptor: ISigAcceptor<SigMessageList>
        ) {
            CallFactory.rpc.create(SigMessageList::class.java, entId, SN, "messageListNext")
                .sendBearerToken()
                .get(msg, sigAcceptor)
        }

        fun messageListPrev(
            entId: EntId,
            msg: MsgMessageListOffset,
            sigAcceptor: ISigAcceptor<SigMessageList>
        ) {
            CallFactory.rpc.create(SigMessageList::class.java, entId, SN, "messageListPrev")
                .sendBearerToken()
                .get(msg, sigAcceptor)
        }

        fun messageMarkRead(entId: EntId, msg: MsgOffset, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "messageMarkRead")
                .sendBearerToken()
                .put(msg, sigAcceptor)
        }

        fun messageMarkReceived(entId: EntId, msg: MsgOffset, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "messageMarkReceived")
                .sendBearerToken()
                .put(msg, sigAcceptor)
        }

        fun messageReactionPut(entId: EntId, msg: MsgReaction, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "messageReactionPut")
                .sendBearerToken()
                .put(msg, sigAcceptor)
        }

        fun messageRemoveForEveryone(
            entId: EntId,
            msg: MsgOffset,
            sigAcceptor: ISigAcceptor<SigDone>
        ) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "messageRemoveForEveryone")
                .sendBearerToken()
                .delete(msg, sigAcceptor)
        }

        fun messageRemoveForMe(entId: EntId, msg: MsgOffset, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "messageRemoveForMe")
                .sendBearerToken()
                .delete(msg, sigAcceptor)
        }

        fun messageReport(entId: EntId, msg: MsgMessageReport, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "messageReport")
                .sendBearerToken()
                .post(msg, sigAcceptor)
        }

        fun messageSend(entId: EntId, msg: MsgMessageSend, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "messageSend")
                .sendBearerToken()
                .post(msg, sigAcceptor)
        }

        fun messageTyping(entId: EntId, msg: MsgChatId, sigAcceptor: ISigAcceptor<SigDone>) {
            CallFactory.rpc.create(SigDone::class.java, entId, SN, "messageTyping")
                .sendBearerToken()
                .post(msg, sigAcceptor)
        }

        fun reverseGeocode(
            entId: EntId,
            msg: MsgReverseGeocode,
            sigAcceptor: ISigAcceptor<SigReverseGeocode>
        ) {
            CallFactory.rpc.create(SigReverseGeocode::class.java, entId, SN, "reverseGeocode")
                .sendBearerToken()
                .get(msg, sigAcceptor)
        }

        fun userLastOnlineGet(
            entId: EntId,
            msg: MsgEntUserIdNoVersion,
            sigAcceptor: ISigAcceptor<SigUserLastOnline>
        ) {
            CallFactory.rpc.create(SigUserLastOnline::class.java, entId, SN, "userLastOnlineGet")
                .sendBearerToken()
                .get(msg, sigAcceptor)
        }

        fun userMessageCandidateListGet(sigAcceptor: ISigAcceptor<SigUserMessageCandidateList>) {
            CallFactory.rpc.create(
                SigUserMessageCandidateList::class.java,
                ApiPlus.ENT_ID_GLOBAL,
                SN,
                "userMessageCandidateListGet"
            )
                .sendBearerToken()
                .get(null, sigAcceptor)
        }
    }
}
